### PR TITLE
Update as per issue #494: updating conversation_tone_analyzer_integration example for Tone Analyzer version 2017-09-21

### LINKED ISSUE #494: Update conversation_tone_analyzer_integration/tone_detection.js to be consistent with Tone Analyzer API v4
Tone Analyzer v4 is removing social and language tones. Demo needs to be updated accordingly.
https://github.ibm.com/people-insights/tone-analyzer/issues/798

### PR DESCRIPTION
**Pull request by aprilwebster**
Fixes #494 

Changes made:
- Updated `tone_detection.js` to be consistent with the latest release (2017-09-21) of The Tone Analyzer:
   - social tones removed; language tones and emotional tones grouped together under `tones`
   - only significant tones (i.e., tones with a score > 0.5 are returned), therefore no longer a need to provide guidance in interpretation in tone_detection.js.
   - comments updated appropriately
- es5 -> es6
- `watson-developer-cloud` package updated to 3.0.1 in `package.json`

Ran example code as per the README (`node tone_conversation_integration.v1.js`)  and the output is as expected: the conversation payload context object contains the expected user object:
```
      "tone": {
        "current": "sadness",
        "history": [
          {
            "tone_name": "sadness",
            "score": 0.962437
          }
        ]
      }
 ```

### CODE REVIEW

File: examples/conversation_tone_analyzer_integration/tone_detection.js [hunk 1]
```diff
@@ -88,175 +67,60 @@ function updateUserTone(conversationPayload, toneAnalyzerPayload, maintainHistor
...
-    });
-
-    updateEmotionTone(user, emotionTone, maintainHistory);
-    updateLanguageTone(user, languageTone, maintainHistory);
-    updateSocialTone(user, socialTone, maintainHistory);
+    updateTone(user, toneAnalyzerPayload.document_tone.tones, maintainHistory);
   }
 
-  conversationPayload.context.user = user;
+  conversationPayload.context.user = user;;
```
Comment (germanattanasio) [inline examples/conversation_tone_analyzer_integration/tone_detection.js]: wasn't one `;` enough?

File: examples/conversation_tone_analyzer_integration/tone_detection.js [hunk 2]
```diff
@@ -88,175 +67,60 @@ function updateUserTone(conversationPayload, toneAnalyzerPayload, maintainHistor
...
 
   return conversationPayload;
 }
 
 /**
  * initToneContext initializes a user object containing tone data (from the Watson Tone Analyzer)
- * @return user json object with the emotion, language and social tones.  The current
- * tone identifies the tone for a specific conversation turn, and the history provides the conversation for
- * all tones up to the current tone for a conversation instance with a user.
+ * @return user json object.  The current tone identifies the tone for a specific conversation turn,
```
Comment (germanattanasio) [inline examples/conversation_tone_analyzer_integration/tone_detection.js]: wasn't one ` ` enough?

File: examples/conversation_tone_analyzer_integration/tone_detection.js (hunk 1 above)
Comment (aprilwebster) [reply examples/conversation_tone_analyzer_integration/tone_detection.js]: removed, thx

File: examples/conversation_tone_analyzer_integration/tone_detection.js (hunk 2 above)
Comment (aprilwebster) [reply examples/conversation_tone_analyzer_integration/tone_detection.js]: 👍

**Pull request merged**